source/material_model/reactive_fluid_transport.cc

```diff
@@ -84,7 +85,11 @@ namespace aspect
               }
               case tian_approximation:
               {
-                melt_fractions[q] = tian2019_model.melt_fraction(in, porosity_idx, q);
```

Comment:
For consistency, I think you should make a similar change in the `zero_solubility` case above. 
Since the idea is that from now on, the material model should take into account the density change upon melting, it would only make sense to consider that the fraction of bound water `in.composition[q][bound_fluid_idx]` has a different density (namely, that of the solid) than the free fluid `in.composition[q][porosity_idx]`, in other words, `in.composition[q][bound_fluid_idx]` should be converted to volume fraction before adding the porosity.